smarts/core/scenario.py

```diff
@@ -97,6 +97,28 @@ def is_reached(self, vehicle):
         return dist <= self.radius
 
 
+@dataclass
+class TraverseGoal(Goal):
+    """An TraverseGoal is satisfied whenever an Agent-driven vehicle
+    successfully finishes traversing a non-closed (acyclical) map
+    It's a way for the vehicle to exit the simulation successfully,
+    for example, driving across from one side to the other on a
+    straight road and then continuing off the map.  This goal is
+    non-specific about *where* the map is exited, save for that
+    the vehicle must be going the correct direction in its lane
+    just prior to doing so."""
+
+    def __init__(self, road_network: SumoRoadNetwork):
+        super().__init__()
+        self._road_network = road_network
+
+    def is_endless(self):
+        return True
+
+    def is_reached(self, vehicle):
```

Comment:
Consider specifying input/output type hints.

Reply:
These match the base class (as I think they should), which doesn't have type hints.   If I change these, I think I should also change the base class, which means changing other classes derived from it.  This is a bit more dangerous.